platform/broadcom/sonic-platform-modules-dell/s6100/sonic_platform/watchdog.py

```diff
@@ -0,0 +1,183 @@
+
+    def get_wd_register(self, reg_name):
+
+        retval = 'ERR'
+        WATCHDOG_SYS_DIR = "/sys/class/watchdog/watchdog0"
+
+        wd_reg_file = WATCHDOG_SYS_DIR+'/'+reg_name
+
+        if (not os.path.isfile(wd_reg_file)):
+            print wd_reg_file,  'not found !'
+            return retval
+
+        try:
+            with open(wd_reg_file, 'r') as fd:
+                retval = fd.read()
+        except Exception as error:
+            logging.error("Unable to open ", wd_reg_file, "file !")
+
+        retval = retval.rstrip('\r\n')
+        retval = retval.lstrip(" ")
+        return retval
+
+    def arm(self, seconds):
+        """
+        Arm the hardware watchdog with a timeout of <seconds> seconds.
+        If the watchdog is currently armed, calling this function will
+        simply reset the timer to the provided value. If the underlying
+        hardware does not support the value provided in <seconds>, this
+        method should arm the watchdog with the *next greater* available
+        value.
+
+        Returns:
+            An integer specifying the *actual* number of seconds the watchdog
+            was armed with. On failure returns -1.
+        """
+
+        # Max timeout is 30 seconds
+        if seconds > 30:
+            seconds = 30
+
+        # Enable watchdog in boot-up
+        watchdog_enable = self.check_config(
+                                self.WATCHDOG_DEFAULT_FILE,
+                                "run_watchdog=1")
+        if watchdog_enable is None:
+            self.write_config(
+                self.WATCHDOG_DEFAULT_FILE,
+                "run_watchdog=0",
+                "run_watchdog=1")
+
+        # configure watchdog-timeout
+        new_timeout = 'watchdog-timeout        = ' + str(seconds)
+        old_timeout = self.check_config(
+                            self.WATCHDOG_CONFIG_FILE,
+                            "watchdog-timeout")
+
+        if old_timeout is not None:
+            self.write_config(
+                self.WATCHDOG_CONFIG_FILE,
+                old_timeout,
+                new_timeout)
+        else:
+            with open(self.WATCHDOG_CONFIG_FILE, "a") as wd_file:
+                wd_file.write(new_timeout)
+            self.run_command(self.WATCHDOG_START)
+
+        # Restart watchdog service
+        self.run_command(self.WATCHDOG_RESTART)
+
+        if self.get_wd_register("timeout") == str(seconds):
+            return seconds
+
+        return -1
+
+    def disarm(self):
+        """
+        Disarm 'watchdog-timeout' in open(self.WATCHDOG_CONFIG_FILE).read():
+         the hardware watchdog
+
+        Returns:
+            A boolean, True if watchdog is disarmed successfully, False if not
+        """
+        if self.get_wd_register("state") == "active":
+
+            # Disable watchdog in boot-up
+            self.write_config(
+                self.WATCHDOG_DEFAULT_FILE,
+                "run_watchdog=1",
+                "run_watchdog=0")
```

Comment:
This will disable the watchdog at next boot. However, this function is meant to disable the watchdog at runtime, in the event there may ever be a need. Is this not possible because of the "nowayout" feature enabled above?

Reply:
Yes. You are right.  The trade-off is with nowayout there is noway to stop watchdog. (Even with magic value)